### PR TITLE
Fix issue with SSA not minimal

### PR DESCRIPTION
**Pull request by sklam**
Fixes https://github.com/numba/numba/issues/5678

### CODE REVIEW

File: numba/core/ssa.py [hunk 1]
```diff
@@ -382,32 +363,31 @@ def _find_def_from_top(self, states, label, loc):
...
-                freshvar = scope.redefine(states['varname'], loc=loc)
-                # insert phi
-                phinode = ir.Assign(
-                    target=freshvar,
-                    value=ir.Expr.phi(loc=loc),
-                    loc=loc,
+        phi_locations = states['phi_locations']
+
+        if label in phi_locations:
+            scope = states['scope']
```
Comment (sklam) [inline numba/core/ssa.py:369]: code in this `if` block is unchanged, just unindent

**Comment by stuartarchibald** [comment]
Thanks for the patch. I've manually checked this, it looks correct. Did you want to consider any of those places in parfors where we had to do `unversioned_name` mangling or get this in and then take a look?

**Comment by stuartarchibald** [comment]
Please could you resolve the conflicts by merging master into this branch. Thanks!

**Comment by sklam** [comment]
RE: https://github.com/numba/numba/pull/5686#issuecomment-629194367

Let's get this in. We will look later.

**Pull request merged**